### PR TITLE
Json config update

### PR DESCRIPTION
**Pull request by sleal-unity**
# Peer Review Information:
This PR updates scenario JSON serialization for DSaaS compatibility. This mainly includes adding support for string, decimal, and boolean scalar values to serialized randomizers and parameters.

## Editor / Package versioning:
**Editor Version Target (i.e. 19.3, 20.1)**: 2019.3

## Dev Testing:
**Tests Added**: 
Updated scenario tests to new format.

## Checklist
- [X] - Updated changelog

### CODE REVIEW

**Review by mkamalza — CHANGES_REQUESTED** [review]
Great work!

File: com.unity.perception/Runtime/Randomization/Scenarios/Scenario.cs [hunk 1]
```diff
@@ -23,202 +17,10 @@ public abstract class Scenario<T> : ScenarioBase where T : ScenarioConstants, ne
...
-#else
-                Debug.Log($"Deserialized scenario configuration from <a href=\"file:///${configFilePath}\">{configFilePath}</a>");
-#endif
-                var jsonText = File.ReadAllText(configFilePath);
-                DeserializeFromJson(jsonText);
-            }
-        }
-
         /// <inheritdoc/>
         public override void DeserializeFromJson(string json)
```
[the remark this answers is not in the document]
Comment (sleal-unity) [reply com.unity.perception/Runtime/Randomization/Scenarios/Scenario.cs:21]: Good catch, I'm axing this.

File: com.unity.perception/Runtime/Randomization/Scenarios/ScenarioBase.cs [hunk 2]
```diff
@@ -132,33 +133,50 @@ protected virtual void IncrementIteration()
...
         /// <summary>
         /// Overwrites this scenario's randomizer settings and scenario constants using a configuration file located at
         /// the provided file path
         /// </summary>
         /// <param name="configFilePath">The file path to the configuration file to deserialize</param>
-        public abstract void DeserializeFromFile(string configFilePath);
+        public virtual void DeserializeFromFile(string configFilePath)
+        {
+            if (string.IsNullOrEmpty(configFilePath) || !File.Exists(configFilePath))
+                Debug.Log($"No configuration file found at {defaultConfigFilePath}");
```
[the remark this answers is not in the document]
Comment (sleal-unity) [reply com.unity.perception/Runtime/Randomization/Scenarios/ScenarioBase.cs:167]: fixed

File: com.unity.perception/Runtime/Randomization/Scenarios/ScenarioBase.cs [hunk 3]
```diff
@@ -132,33 +133,50 @@ protected virtual void IncrementIteration()
...
+            else
+            {
+#if UNITY_EDITOR
+                Debug.Log($"Deserialized scenario configuration from <a href=\"file:///${configFilePath}\">{configFilePath}</a>. " +
```
[the remark this answers is not in the document]
Comment (sleal-unity) [reply com.unity.perception/Runtime/Randomization/Scenarios/ScenarioBase.cs:171]: Moved this logging code under the actual serialization as suggested

File: com.unity.perception/Runtime/Randomization/Scenarios/Serialization/ScenarioSerializer.cs [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+            if (field.FieldType == typeof(float) || field.FieldType == typeof(double) || field.FieldType == typeof(int))
+                return new DoubleScalarValue { num = Convert.ToDouble(field.GetValue(obj)) };
+            return null;
+        }
+        #endregion
+
+        #region Deserialization
+        public static void Deserialize(ScenarioBase scenario, string json)
+        {
+            var jsonData = JObject.Parse(json);
```
[the remark this answers is not in the document]
Comment (sleal-unity) [reply com.unity.perception/Runtime/Randomization/Scenarios/Serialization/ScenarioSerializer.cs:161]: Addressed in latest update

**Review by mkamalza — APPROVED** [review]
🟢

**Pull request merged**